src/Language/Futhark/Primitive.hs

```diff
@@ -321,6 +322,11 @@ floatValueType Float16Value {} = Float16
 floatValueType Float32Value {} = Float32
 floatValueType Float64Value {} = Float64
 
+valueRational :: FloatValue -> Rational
```

Comment:
This is dangerous, and it is highly likely that whatever code that uses this is wrong in edge cases.